src/lib/status-manager.ts

```diff
@@ -0,0 +1,97 @@
+import type {ExecaChildProcess, ExecaReturnValue} from 'execa';
+import type {Socket} from 'socket.io-client';
+import parse from '../command/handlers/ping/parse.js';
+import type {PingOptions} from '../command/ping-command.js';
+import {hasRequired} from './dependencies.js';
+import {scopedLogger} from './logger.js';
+
+const logger = scopedLogger('status-manager');
+
+const INTERVAL_TIME = 10 * 60 * 1000; // 10 mins
+
+export class StatusManager {
+	private status: 'initializing' | 'ready' | 'unbuffer-missing' | 'ping-test-failed' | 'sigterm' = 'initializing';
+	private timer?: NodeJS.Timeout;
+
+	constructor(
+		private readonly socket: Socket,
+		private readonly pingCmd: (options: PingOptions) => ExecaChildProcess,
+	) {}
+
+	public async start() {
+		const hasRequiredDeps = await hasRequired();
+
+		if (!hasRequiredDeps) {
+			this.updateStatus('unbuffer-missing');
+			return;
+		}
+
+		await this.runTest();
+	}
+
+	public stop(status: StatusManager['status']) {
+		this.updateStatus(status);
+		clearTimeout(this.timer);
+	}
+
+	public getStatus() {
+		return this.status;
+	}
+
+	public updateStatus(status: StatusManager['status']) {
+		if (status !== this.status) {
+			this.status = status;
+			this.sendStatus();
+		}
+	}
+
+	public sendStatus() {
+		this.socket.emit('probe:status:update', this.status);
+	}
+
+	private async runTest() {
+		const result = await this.pingTest();
+		if (result) {
+			this.updateStatus('ready');
+		} else {
+			this.updateStatus('ping-test-failed');
+		}
+
+		this.timer = setTimeout(async () => {
+			await this.runTest();
+		}, INTERVAL_TIME);
```

Comment:
Is this meant to be `setInterval()` or where do we schedule the next run?

Reply:
runTest recursively calls next runTest

Reply:
🤦‍♂️ I got it confused with `pingTest()`